### ISSUE TITLE
Error: Compiled vertex/fragment shader was corrupt.

### ISSUE THREAD

**Issue by datatypevoid** [nickgildea/DualContouringSample#2]
Attempting to run the program in macOS Sierra yields the following output:

```
/Users/da5id/Projects/tutorials/DualContouringSample/cmake-build-debug/DualContourSample
----------------------------------------------------------------
The controls are:
	- hold left mouse and drag to rotate the view
	- use the mouse wheel to zoom in/out
	- press F1 to render a wireframe
	- press F2 to regenerate the octree using a new error threshold (and generate a new mesh)
----------------------------------------------------------------


OpenGL version: 4.1 INTEL-10.20.25
OpenGL shading version: 4.10
ERROR: Compiled vertex shader was corrupt.
ERROR: Compiled fragment shader was corrupt.

Error: failed to create GLSL program

Process finished with exit code 1
```

I am using `CLion 2016.3` as my IDE.

I read online in a few places that there could be cross-platform encoding problems, but using `dos2unix` or even manually rewriting the shaders has not yielded any progress. Any insight would be awesome! 

On another note, thank you for such information-rich blog posts, they are invaluable along the journey of rendering smooth voxel terrain.

**Comment by nickgildea** [comment]
Hi, I'm glad you found the posts helpful.

I've not tried the code on MacOS myself. I would see if you can get some more debugging info out of OpenGL by adding some more debugging calls (e.g. Maybe to glGetError or similar?)

The other thing to try would be gutting the shaders to the absolute minimum and then gradually adding in more code until you get the error again, and see if you can 1) identify the problematic but of code, and 2) try and fix it.

I've ran the code on a couple of windows systems and on Nvidia and AMD cards so I would suspect it's perhaps a MacOS problem of some sort, maybe not supporting the version of OpenGL required?

Good luck!

**Comment by datatypevoid** [comment]
Thanks for the quick response!

I had a feeling that it might be a macOS specific issue. In that case, I'll try it on my Windows box first. I just need to see the running program in order to fully understand the code I am looking at. I will poke some more at the macOS issue in the coming days as that is my primary development box, and keep you updated if I figure out what is wrong.

**Comment by datatypevoid** [comment]
It struck me today that the build system was actually placing the executable in a separate directory. In order to resolve this issue, I wrote a quick script to copy the shader files to the appropriate directory upon building the project. I also had to rewrite the shader filers to work with the newer versions of OpenGL, but that was only a few trivial changes. If you would like I can submit a PR with the updated shader files, otherwise, thank you again for the wealth of information you have made freely available. I look forward to attempting to make the changes listed in the subsequent blog posts.
